src/app/(post-login)/submission/components/ProgramMenu.tsx

```diff
@@ -71,7 +71,7 @@ const ProgramMenu = ({ shortNameSearchQuery }: { shortNameSearchQuery: string })
 	const { DATA_CENTER } = useAppConfigContext();
 
 	// params can be arrays from dynamic routing
-	const activeProgramName = typeof params.shortName === 'string' ? params.shortName : '';
+	let activeProgramName = typeof params.shortName === 'string' ? params.shortName : '';
```

Comment:
`const` vs `let` in react world is a code smell to me because there's not a whole lot of uses cases for `let` in react.
using `let` here implies we want this value to change which implies it's state and UI reacting to said state.

because React will batch and reconcile state changes internally to write to it's virtual DOM we have no control over it if we use let, const .. var.. in short. React might know about the change - but it might not. I don't know if React just ignores "state like variables".

because we're changing the `activeProgramName` in a click handler to change a `let`, we haven't told React it's happening. we should use React.useState

for the sake of covering all the bases here - there are still cases to use regular variables! derived values are a great example. here I'm specifically commenting on this because we are using it as state that pieces of UI depend on

@joneubank @justincorrigible  have some more experience than me on this and may be able to explain it better ( I could also be straight up incorrect ;) )

Following on from my earlier comment I think we should just leave this activeProgram side menu bug for it's own isolated ticket/PR. Can we revert the changes to the sidemenu in this PR so it's not merged in with the other work?

Reply:
You're correct and I agree, this is merely a side effect of not fixing the Side Menu open/close bug as part of this ticket, which would require introducing component state in addition to the URL state.

I will use `const activeProgramName` and the bug remains for now, and fix the rest in a separate branch.